src/server/services/wms/qgswmsrestorer.cpp

```diff
@@ -74,18 +74,19 @@ QgsLayerRestorer::~QgsLayerRestorer()
 {
   for ( QgsMapLayer *layer : mLayerSettings.keys() )
   {
-    const QgsLayerSettings settings = mLayerSettings[layer];
-    layer->styleManager()->setCurrentStyle( settings.mNamedStyle );
-    layer->setName( mLayerSettings[layer].name );
-
-    // if a SLD file has been loaded for rendering, we restore the previous style
+    // Firstly check if a SLD file has been loaded for rendering and removed it
     const QString sldStyleName { layer->customProperty( "sldStyleName", "" ).toString() };
     if ( !sldStyleName.isEmpty() )
     {
       layer->styleManager()->removeStyle( sldStyleName );
       layer->removeCustomProperty( "sldStyleName" );
     }
 
+    // Then restore the previous style
+    const QgsLayerSettings settings = mLayerSettings[layer];
+    layer->styleManager()->setCurrentStyle( settings.mNamedStyle );
+    layer->setName( mLayerSettings[layer].name );
```

Comment:
Do you know why the order is important?

Reply:
Yes, the custom variables are stored in the style, so the style has to be reset after getting the variables.

Reply:
ok, thanks for the explanation :+1: